pytensor/tensor/rewriting/subtensor.py

```diff
@@ -381,6 +381,51 @@ def local_useless_slice(fgraph, node):
             return [node.inputs[0]]
 
 
+@register_infer_shape
+@register_canonicalize
+@register_specialize
```

Comment:
```suggestion
@register_useless
@register_canonicalize
@register_stabilize
@register_specialize
```
Not clear this is needed for infer_shape, but it should be in all the main passes otherwise